### PR TITLE
Abandoned calls

### PR DESCRIPTION
**Pull request by benlangfeld**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -90,6 +90,12 @@ agent = ElectricSlide::Agent.new options
 ElectricSlide.update_agent 1, agent
 ```
 
+The possible presence states an Agent may be in are:
+
+* `:available` - Waiting for a call
+* `:on_call` - Currently connected to a call
+* `:after_call` - In a quiet period after completing a call and before being made available again. This is only encountered with a manual agent return strategy.
```
Comment (bklang) [inline README.markdown]: We should also mention `:unavailable`. I realize this means "remove from the queue", but it's still a valid state, one that can be passed into the API.

File: lib/electric_slide/call_queue.rb [hunk 2]
```diff
@@ -151,7 +151,8 @@ def check_for_connections
     # @param [Adhearsion::Call] call Caller to be added to the queue
     def priority_enqueue(call)
       # Don't reset the enqueue time in case this is a re-insert on agent failure
-      call[:enqueue_time] ||= Time.now
+      call[:electric_slide_enqueued_at] ||= DateTime.now
```
Comment (bklang) [inline lib/electric_slide/call_queue.rb]: This should be noted in the CHANGELOG

File: lib/electric_slide/agent.rb [hunk 3]
```diff
@@ -5,11 +5,11 @@ class ElectricSlide::Agent
   # @param [Hash] opts Agent parameters
   # @option opts [String] :id The Agent's ID
   # @option opts [String] :address The Agent's contact address
-  # @option opts [Symbol] :presence The Agent's current presence. Must be one of :available, :on_call, :away, :offline
+  # @option opts [Symbol] :presence The Agent's current presence. Must be one of :available, :on_call, :after_call, :unavailable
```
Comment (bklang) [inline lib/electric_slide/agent.rb]: This should be noted in the CHANGELOG

File: lib/electric_slide/call_queue.rb [hunk 4]
```diff
@@ -151,7 +151,8 @@ def check_for_connections
...
+      call.on_end { abandon call }
```
Comment (bklang) [inline lib/electric_slide/call_queue.rb]: If we're going to use `#abandon` to mean any ending of the call, whether before connecting to an agent or after, we need to rename this method. For example, the log message inside `#abandon` logs the fact that it is an abandon. Abandon in this context specifically means "left the queue before getting to an agent."  This is an important fact to track, so we should probably update the method to check whether a connection occurred, and log an appropriate message.

File: README.markdown (hunk 1 above)
Comment (benlangfeld) [inline README.markdown]: Wrong PR

Comment (benlangfeld) [inline README.markdown]: This is not a valid state. Cannot be passed to this API for any purpose.

File: lib/electric_slide/agent.rb (hunk 3 above)
Comment (benlangfeld) [inline lib/electric_slide/agent.rb]: This project does not have a changelog.

**Comment by benlangfeld** [comment]
Review please @bklang.

**Comment by bklang** [comment]
:ship:

**Pull request merged**